homeassistant/components/smlight/binary_sensor.py

```diff
@@ -78,3 +89,47 @@ def __init__(
     def is_on(self) -> bool:
         """Return the state of the sensor."""
         return self.entity_description.value_fn(self.coordinator.data.sensors)
+
+
+class SmInternetSensorEntity(SmEntity, BinarySensorEntity):
+    """Representation of the SLZB internet sensor."""
+
+    def __init__(
+        self,
+        coordinator: SmDataUpdateCoordinator,
+    ) -> None:
+        """Initialize slzb binary sensor."""
+        super().__init__(coordinator)
+        self._attr_name = "Internet"
+        self._attr_translation_key = "internet"
+        self._attr_unique_id = f"{coordinator.unique_id}_{self._attr_name}"
+        self._attr_device_class = BinarySensorDeviceClass.CONNECTIVITY
+        self._attr_entity_category = EntityCategory.DIAGNOSTIC
```

Comment:
Everything except for the unique_id can be set outside of the cosntructor

Reply:
ok